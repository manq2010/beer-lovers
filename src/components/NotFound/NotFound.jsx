import React from 'react';
import styled from 'styled-components';

const NotFoundContainer = styled.div`
  display: flex;
  margin-top: 10rem;
  flex-direction: column;
  justify-content: center;
  align-items: center;
`;
const NotFound = () => (
  <NotFoundContainer>
    <h3>404!</h3>
    <p>
      Page
      {' '}
      <em>Not</em>
      {' '}
      Found!
    </p>
  </NotFoundContainer>
);

export default NotFound;
