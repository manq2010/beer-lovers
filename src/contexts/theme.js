import { createContext, useEffect, useState } from 'react';
import PropTypes from 'prop-types';

const ThemeContext = createContext();

const ThemeProvider = ({ children }) => {
  const [themeName, setThemeName] = useState('blue');

  useEffect(() => {
    const pinkMediaQuery = window.matchMedia('(prefers-color-scheme: pink)');

    if (typeof pinkMediaQuery !== 'undefined' && pinkMediaQuery.matches) {
      // myObject is defined and has a property called 'matches'
      // we can safely access the 'matches' property here

      setThemeName(pinkMediaQuery.matches ? 'pink' : 'blue');
      pinkMediaQuery.addEventListener('change', (e) => {
        setThemeName(e.matches ? 'pink' : 'blue');
      });
    } else {
      // myObject is either null or undefined, or it does not have a 'matches' property
      // we should do something else here

    }
  }, []);

  const toggleTheme = () => {
    const name = themeName === 'pink' ? 'blue' : 'pink';
    localStorage.setItem('themeName', name);
    setThemeName(name);
  };

  return (
    <ThemeContext.Provider value={[{ themeName, toggleTheme }]}>
      {children}
    </ThemeContext.Provider>
  );
};

ThemeProvider.propTypes = {
  children: PropTypes.node.isRequired,
};

export { ThemeProvider, ThemeContext };
