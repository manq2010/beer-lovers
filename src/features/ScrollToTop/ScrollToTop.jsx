import React, { useState } from 'react';
import ArrowCircleUpIcon from '@mui/icons-material/ArrowCircleUp';

const ScrollToTop = () => {
  const [visible, setVisible] = useState(false);

  const toggleVisible = () => {
    const scrolled = document.documentElement.scrollTop;
    if (scrolled > 500) {
      setVisible(true);
    } else if (scrolled <= 500) {
      setVisible(false);
    }
  };

  const backToTop = () => {
    window.scrollTo({
      top: 0,
      behavior: 'smooth',
    });
  };

  window.addEventListener('scroll', toggleVisible);

  return (
    <div
      style={{
        display: visible ? 'inline' : 'none',
        position: 'fixed',
        right: '1rem',
        bottom: '3.5rem',
        height: '3rem',
        zIndex: '10',
      }}
    >
      <button
        className="btn-scroll"
        style={{
          outline: 'none',
          border: 'none',
          background: 'none',
          cursor: 'pointer',
        }}
        type="button"
        onClick={backToTop}
        aria-label="Back to top"
      >
        <ArrowCircleUpIcon style={{ fontSize: '2.5rem' }} />
      </button>
    </div>
  );
};

export default ScrollToTop;
