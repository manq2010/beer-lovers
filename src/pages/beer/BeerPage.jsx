import React from 'react';
import { Helmet } from 'react-helmet';
import { useLocation } from 'react-router-dom';
import headerData from '../../data/headerData';
import SingleBeer from '../../components/Beers/SingleBeer/SingleBeer';
import Contact from '../../components/Contact/Contact';
import Footer from '../../components/Footer/Footer';
import Navbar from '../../components/Navbar/Navbar';
import shortName from '../../utils/shortName';

const BeerPage = () => {
  const location = useLocation();
  // if (location.state) {
  //   const { stateData } = location.state;
  // }
  const stateData = location.state?.stateData;

  return (
    <>
      <Helmet>
        <title>
          {stateData.beer.beer_name}
          {' '}
          Beer
          {' '}
          |
          {' '}
          {shortName(headerData.name, 10)}
          {' '}
          |
          {' '}
          Beer Lovers App
        </title>
        <meta name="description" content="Beer lovers application" />
      </Helmet>
      <Navbar />
      <div>
        {
          stateData && typeof stateData === 'object' && (
            <SingleBeer
              key={stateData.beer.id}
              beer={stateData.beer}
            />
          )
        }
      </div>
      <Contact />
      <Footer />
    </>
  );
};

export default BeerPage;
