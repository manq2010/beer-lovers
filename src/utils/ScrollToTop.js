/* eslint-disable react/prop-types */
import React, { useEffect } from 'react';
import { withRouter } from 'react-router-dom';

const ScrollToTop = ({ history }) => {
  useEffect(() => {
    const unlisten = history.listen(() => {
      window.scrollTo(0, 0);
    });
    return () => {
      unlisten();
    };
  });
  return (
    <div />
  );
};

// export default withRouter(ScrollToTop);
