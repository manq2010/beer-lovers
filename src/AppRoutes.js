import React from 'react';
import { Route, Routes } from 'react-router-dom';
// Pages
import NotFoundPage from './pages/notFound/NotFoundPage';
import MainPage from './pages/main/MainPage';
import BeerPage from './pages/beer/BeerPage';

const AppRoutes = () => (
  <Routes>
    <Route path="/" element={<MainPage />} />
    <Route path="*" element={<NotFoundPage />} />
    <Route path="/beer" element={<BeerPage />} />
  </Routes>
);

export default AppRoutes;
