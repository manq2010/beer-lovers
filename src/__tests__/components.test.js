import { BrowserRouter, Router } from 'react-router-dom';
import { createMemoryHistory } from 'history';
import { render, screen, fireEvent } from '@testing-library/react';
import renderer from 'react-test-renderer';
import { Provider } from 'react-redux';
// import ReactTestUtils, { act } from 'react-dom/test-utils';
import store from '../redux/configureStore';
import { ThemeProvider } from '../contexts/theme';
import AppRoutes from '../AppRoutes';
import NotFound from '../components/NotFound/NotFound';
import MainPage from '../pages/main/MainPage';
import BeerPage from '../pages/beer/BeerPage';
import ScrollToTop from '../components/ScrollToTop/ScrollToTop';
// import App from '../App';

Object.defineProperty(window, 'matchMedia', {
  writable: true,
  value: jest.fn().mockImplementation((query) => ({
    matches: false,
    media: query,
    onchange: null,
    addListener: jest.fn(), // deprecated
    removeListener: jest.fn(), // deprecated
    addEventListener: jest.fn(),
    removeEventListener: jest.fn(),
    dispatchEvent: jest.fn(),
  })),
});

// Routing testing

describe('Navbar renders', () => {
  test('should redirect and update history', () => {
    const history = createMemoryHistory();
    render(
      <Provider store={store}>
        <ThemeProvider>
          <Router location={history.location} navigator={history}>
            <AppRoutes />
          </Router>
        </ThemeProvider>
      </Provider>,
    );

    fireEvent.click(screen.getByTestId(/Beer Lovers/));
    expect(history.location.pathname).toEqual('/');
  });

  test('should redirect and update dom', () => {
    render(
      <Provider store={store}>
        <ThemeProvider>
          <BrowserRouter>
            <AppRoutes />
          </BrowserRouter>
        </ThemeProvider>
      </Provider>,
    );

    fireEvent.click(screen.getByTestId(/Beer Lovers/));
    expect(screen.getByText(/Loading.../)).toBeInTheDocument();
  });
});

// Component testing

describe('Not-found page is rendered', () => {
  test('Component is rendered', () => {
    render(
      <Provider store={store}>
        <ThemeProvider>
          <NotFound />
        </ThemeProvider>
      </Provider>,
    );
    const container = screen.getByText(/404!/);
    expect(container).toBeInTheDocument();
  });

  test('matches snapshot', () => {
    const tree = renderer.create(
      <Provider store={store}>
        <ThemeProvider>
          <NotFound />
        </ThemeProvider>
      </Provider>,
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });
});

describe('MainPage component is rendererd', () => {
  test('Component is rendererd', () => {
    render(
      <Provider store={store}>
        <ThemeProvider>
          <BrowserRouter>
            <MainPage />
          </BrowserRouter>
        </ThemeProvider>
      </Provider>,
    );
    const container = screen.getByText(/Beer Lovers/);
    expect(container).toBeInTheDocument();
  });

  test('matches snapshot', () => {
    const tree = renderer.create(
      <Provider store={store}>
        <ThemeProvider>
          <BrowserRouter>
            <MainPage />
          </BrowserRouter>
        </ThemeProvider>
      </Provider>,
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });
});

describe('BeerPage component is rendererd', () => {
  test('Component is rendererd with data from API', () => {
    render(
      <Provider store={store}>
        <ThemeProvider>
          <BrowserRouter>
            <BeerPage />
          </BrowserRouter>
        </ThemeProvider>
      </Provider>,
    );
    const container = screen.queryByText(/First Brew/);
    expect(container).toBeNull();
  });

  test('matches snapshot', () => {
    const tree = renderer.create(
      <Provider store={store}>
        <ThemeProvider>
          <BrowserRouter>
            <BeerPage />
          </BrowserRouter>
        </ThemeProvider>
      </Provider>,
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });
});

describe('ScrollToTop component is rendererd', () => {
// Mock for testing purposes
  const mockScrollTo = jest.fn();
  window.scrollTo = mockScrollTo;

  test('Component is rendererd', () => {
    render(
      <ScrollToTop />,
    );

    fireEvent.click(screen.getByTestId(/scroll/));
    expect(mockScrollTo).toHaveBeenCalledWith({ top: 0, behavior: 'smooth' });
  });
});
