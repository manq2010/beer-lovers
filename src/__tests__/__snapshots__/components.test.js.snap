// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BeerPage component is rendererd matches snapshot 1`] = `
Array [
  <nav>
    <ul
      style={
        Object {
          "alignItems": "center",
          "display": "flex",
          "justifyContent": "space-between",
        }
      }
    >
      <li>
        <a
          aria-current="page"
          className="active"
          data-testid="Beer Lovers"
          href="/"
          onClick={[Function]}
          style={
            Object {
              "color": "#fff",
              "textDecoration": "none",
            }
          }
        >
          <h1
            style={
              Object {
                "alignItems": "center",
                "display": "flex",
                "gap": "0.5rem",
                "justifyContent": "space-between",
                "margin": "1rem 0 0 1rem",
              }
            }
          >
            <svg
              aria-hidden={true}
              className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
              data-testid="SportsBarIcon"
              focusable="false"
              style={
                Object {
                  "fontSize": "2.5rem",
                }
              }
              viewBox="0 0 24 24"
            >
              <path
                d="M19 9h-1.56c.35-.59.56-1.27.56-2 0-2.21-1.79-4-4-4-.34 0-.66.05-.98.13-.82-.68-1.86-1.11-3.02-1.11-1.89 0-3.51 1.11-4.27 2.71C4.15 5.26 3 6.74 3 8.5c0 1.86 1.28 3.41 3 3.86V21h11v-2h2c1.1 0 2-.9 2-2v-6c0-1.1-.9-2-2-2zM7 10.5c-1.1 0-2-.9-2-2 0-.85.55-1.6 1.37-1.88l.8-.27.36-.76C8 4.62 8.94 4.02 10 4.02c.79 0 1.39.35 1.74.65l.78.65S13.16 5 13.99 5c1.1 0 2 .9 2 2h-3C9.67 7 9.15 10.5 7 10.5zM19 17h-2v-6h2v6z"
              />
            </svg>
            <span
              className="logo"
            >
              Beer Lovers
            </span>
          </h1>
        </a>
      </li>
      <div
        style={
          Object {
            "alignItems": "center",
            "display": "flex",
            "justifyContent": "center",
          }
        }
      >
        <div>
          <li>
            <button
              className="search-button"
              onClick={[Function]}
              style={
                Object {
                  "alignItems": "center",
                  "backgroundColor": "transparent",
                  "border": "1px solid #fff",
                  "borderRadius": "5px",
                  "color": "#fff",
                  "cursor": "pointer",
                  "display": "flex",
                  "gap": "0.5rem",
                  "justifyContent": "space-between",
                  "margin": "1rem 0.5rem 0 0",
                  "paddingRight": "0.5rem",
                }
              }
              type="button"
            >
              <svg
                aria-hidden={true}
                className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                data-testid="SearchIcon"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M15.5 14h-.79l-.28-.27C15.41 12.59 16 11.11 16 9.5 16 5.91 13.09 3 9.5 3S3 5.91 3 9.5 5.91 16 9.5 16c1.61 0 3.09-.59 4.23-1.57l.27.28v.79l5 4.99L20.49 19l-4.99-5zm-6 0C7.01 14 5 11.99 5 9.5S7.01 5 9.5 5 14 7.01 14 9.5 11.99 14 9.5 14z"
                />
              </svg>
               
              <span
                className="logo"
              >
                Search...
              </span>
            </button>
          </li>
        </div>
        <li>
          <button
            aria-label="toggle theme"
            onClick={[Function]}
            style={
              Object {
                "alignItems": "center",
                "backgroundColor": "transparent",
                "border": "none",
                "display": "flex",
                "justifyContent": "center",
                "margin": "1rem 1rem 0 0",
                "outline": "none",
              }
            }
            type="button"
          >
            <svg
              aria-hidden={true}
              className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium theme-btn css-i4bv87-MuiSvgIcon-root"
              data-testid="ToggleOnIcon"
              focusable="false"
              style={
                Object {
                  "fontSize": "2.7rem",
                }
              }
              viewBox="0 0 24 24"
            >
              <path
                d="M17 7H7c-2.76 0-5 2.24-5 5s2.24 5 5 5h10c2.76 0 5-2.24 5-5s-2.24-5-5-5zm0 8c-1.66 0-3-1.34-3-3s1.34-3 3-3 3 1.34 3 3-1.34 3-3 3z"
              />
            </svg>
          </button>
        </li>
      </div>
    </ul>
  </nav>,
  <div />,
  <div
    style={
      Object {
        "color": "#fff",
        "margin": "1rem 0 1rem 0",
        "textTransform": "capitalize",
      }
    }
  >
    <h2
      className="center"
    >
      Connect with me
    </h2>
    <div
      style={
        Object {
          "alignItems": "center",
          "display": "flex",
          "gap": "0.5rem",
          "justifyContent": "center",
          "marginTop": "1rem",
        }
      }
    >
      <a
        href="https://twitter.com/mancoba_c"
        rel="noreferrer"
        target="_blank"
      >
        <svg
          aria-hidden={true}
          aria-label="Twitter"
          className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
          data-testid="TwitterIcon"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M22.46 6c-.77.35-1.6.58-2.46.69.88-.53 1.56-1.37 1.88-2.38-.83.5-1.75.85-2.72 1.05C18.37 4.5 17.26 4 16 4c-2.35 0-4.27 1.92-4.27 4.29 0 .34.04.67.11.98C8.28 9.09 5.11 7.38 3 4.79c-.37.63-.58 1.37-.58 2.15 0 1.49.75 2.81 1.91 3.56-.71 0-1.37-.2-1.95-.5v.03c0 2.08 1.48 3.82 3.44 4.21a4.22 4.22 0 0 1-1.93.07 4.28 4.28 0 0 0 4 2.98 8.521 8.521 0 0 1-5.33 1.84c-.34 0-.68-.02-1.02-.06C3.44 20.29 5.7 21 8.12 21 16 21 20.33 14.46 20.33 8.79c0-.19 0-.37-.01-.56.84-.6 1.56-1.36 2.14-2.23z"
          />
        </svg>
      </a>
      <a
        href="https://github.com/manq2010"
        rel="noreferrer"
        target="_blank"
      >
        <svg
          aria-hidden={true}
          aria-label="github"
          className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
          data-testid="GitHubIcon"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M12 1.27a11 11 0 00-3.48 21.46c.55.09.73-.28.73-.55v-1.84c-3.03.64-3.67-1.46-3.67-1.46-.55-1.29-1.28-1.65-1.28-1.65-.92-.65.1-.65.1-.65 1.1 0 1.73 1.1 1.73 1.1.92 1.65 2.57 1.2 3.21.92a2 2 0 01.64-1.47c-2.47-.27-5.04-1.19-5.04-5.5 0-1.1.46-2.1 1.2-2.84a3.76 3.76 0 010-2.93s.91-.28 3.11 1.1c1.8-.49 3.7-.49 5.5 0 2.1-1.38 3.02-1.1 3.02-1.1a3.76 3.76 0 010 2.93c.83.74 1.2 1.74 1.2 2.94 0 4.21-2.57 5.13-5.04 5.4.45.37.82.92.82 2.02v3.03c0 .27.1.64.73.55A11 11 0 0012 1.27"
          />
        </svg>
      </a>
      <a
        href="https://www.linkedin.com/in/mancoba-sihlongonyane/"
        rel="noreferrer"
        target="_blank"
      >
        <svg
          aria-hidden={true}
          aria-label="LinkedIn"
          className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
          data-testid="LinkedInIcon"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M19 3a2 2 0 0 1 2 2v14a2 2 0 0 1-2 2H5a2 2 0 0 1-2-2V5a2 2 0 0 1 2-2h14m-.5 15.5v-5.3a3.26 3.26 0 0 0-3.26-3.26c-.85 0-1.84.52-2.32 1.3v-1.11h-2.79v8.37h2.79v-4.93c0-.77.62-1.4 1.39-1.4a1.4 1.4 0 0 1 1.4 1.4v4.93h2.79M6.88 8.56a1.68 1.68 0 0 0 1.68-1.68c0-.93-.75-1.69-1.68-1.69a1.69 1.69 0 0 0-1.69 1.69c0 .93.76 1.68 1.69 1.68m1.39 9.94v-8.37H5.5v8.37h2.77z"
          />
        </svg>
      </a>
      <a
        href="mailto:manq2010@gmail.com"
        rel="noreferrer"
        target="_blank"
      >
        <span
          type="button"
        >
          <svg
            aria-hidden={true}
            className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
            data-testid="EmailIcon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M20 4H4c-1.1 0-1.99.9-1.99 2L2 18c0 1.1.9 2 2 2h16c1.1 0 2-.9 2-2V6c0-1.1-.9-2-2-2zm0 4-8 5-8-5V6l8 5 8-5v2z"
            />
          </svg>
        </span>
      </a>
    </div>
  </div>,
  <div
    style={
      Object {
        "alignItems": "center",
        "borderTop": "1px solid #fff",
        "display": "flex",
        "gap": "1rem",
        "justifyContent": "space-around",
      }
    }
  >
    <div
      style={
        Object {
          "margin": "1rem 0 1rem 0",
        }
      }
    >
      <p>
        Created by
         
        <em>
          Mancoba Sihlongonyane
        </em>
      </p>
    </div>
    <div>
      <p>
        Â© Copyright
         
        2023 -
         
        2023
      </p>
    </div>
  </div>,
]
`;

exports[`MainPage component is rendererd matches snapshot 1`] = `
<div>
  <nav>
    <ul
      style={
        Object {
          "alignItems": "center",
          "display": "flex",
          "justifyContent": "space-between",
        }
      }
    >
      <li>
        <a
          aria-current="page"
          className="active"
          data-testid="Beer Lovers"
          href="/"
          onClick={[Function]}
          style={
            Object {
              "color": "#fff",
              "textDecoration": "none",
            }
          }
        >
          <h1
            style={
              Object {
                "alignItems": "center",
                "display": "flex",
                "gap": "0.5rem",
                "justifyContent": "space-between",
                "margin": "1rem 0 0 1rem",
              }
            }
          >
            <svg
              aria-hidden={true}
              className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
              data-testid="SportsBarIcon"
              focusable="false"
              style={
                Object {
                  "fontSize": "2.5rem",
                }
              }
              viewBox="0 0 24 24"
            >
              <path
                d="M19 9h-1.56c.35-.59.56-1.27.56-2 0-2.21-1.79-4-4-4-.34 0-.66.05-.98.13-.82-.68-1.86-1.11-3.02-1.11-1.89 0-3.51 1.11-4.27 2.71C4.15 5.26 3 6.74 3 8.5c0 1.86 1.28 3.41 3 3.86V21h11v-2h2c1.1 0 2-.9 2-2v-6c0-1.1-.9-2-2-2zM7 10.5c-1.1 0-2-.9-2-2 0-.85.55-1.6 1.37-1.88l.8-.27.36-.76C8 4.62 8.94 4.02 10 4.02c.79 0 1.39.35 1.74.65l.78.65S13.16 5 13.99 5c1.1 0 2 .9 2 2h-3C9.67 7 9.15 10.5 7 10.5zM19 17h-2v-6h2v6z"
              />
            </svg>
            <span
              className="logo"
            >
              Beer Lovers
            </span>
          </h1>
        </a>
      </li>
      <div
        style={
          Object {
            "alignItems": "center",
            "display": "flex",
            "justifyContent": "center",
          }
        }
      >
        <div>
          <li>
            <button
              className="search-button"
              onClick={[Function]}
              style={
                Object {
                  "alignItems": "center",
                  "backgroundColor": "transparent",
                  "border": "1px solid #fff",
                  "borderRadius": "5px",
                  "color": "#fff",
                  "cursor": "pointer",
                  "display": "flex",
                  "gap": "0.5rem",
                  "justifyContent": "space-between",
                  "margin": "1rem 0.5rem 0 0",
                  "paddingRight": "0.5rem",
                }
              }
              type="button"
            >
              <svg
                aria-hidden={true}
                className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                data-testid="SearchIcon"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M15.5 14h-.79l-.28-.27C15.41 12.59 16 11.11 16 9.5 16 5.91 13.09 3 9.5 3S3 5.91 3 9.5 5.91 16 9.5 16c1.61 0 3.09-.59 4.23-1.57l.27.28v.79l5 4.99L20.49 19l-4.99-5zm-6 0C7.01 14 5 11.99 5 9.5S7.01 5 9.5 5 14 7.01 14 9.5 11.99 14 9.5 14z"
                />
              </svg>
               
              <span
                className="logo"
              >
                Search...
              </span>
            </button>
          </li>
        </div>
        <li>
          <button
            aria-label="toggle theme"
            onClick={[Function]}
            style={
              Object {
                "alignItems": "center",
                "backgroundColor": "transparent",
                "border": "none",
                "display": "flex",
                "justifyContent": "center",
                "margin": "1rem 1rem 0 0",
                "outline": "none",
              }
            }
            type="button"
          >
            <svg
              aria-hidden={true}
              className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium theme-btn css-i4bv87-MuiSvgIcon-root"
              data-testid="ToggleOnIcon"
              focusable="false"
              style={
                Object {
                  "fontSize": "2.7rem",
                }
              }
              viewBox="0 0 24 24"
            >
              <path
                d="M17 7H7c-2.76 0-5 2.24-5 5s2.24 5 5 5h10c2.76 0 5-2.24 5-5s-2.24-5-5-5zm0 8c-1.66 0-3-1.34-3-3s1.34-3 3-3 3 1.34 3 3-1.34 3-3 3z"
              />
            </svg>
          </button>
        </li>
      </div>
    </ul>
  </nav>
  <div>
    <h3
      style={
        Object {
          "alignItems": "center",
          "display": "flex",
          "height": "65vh",
          "justifyContent": "center",
        }
      }
    >
      <span>
        <svg
          aria-hidden={true}
          className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
          data-testid="HourglassTopIcon"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="m6 2 .01 6L10 12l-3.99 4.01L6 22h12v-6l-4-4 4-3.99V2H6zm10 14.5V20H8v-3.5l4-4 4 4z"
          />
        </svg>
      </span>
       
      Loading...
    </h3>
  </div>
  <div
    style={
      Object {
        "color": "#fff",
        "margin": "1rem 0 1rem 0",
        "textTransform": "capitalize",
      }
    }
  >
    <h2
      className="center"
    >
      Connect with me
    </h2>
    <div
      style={
        Object {
          "alignItems": "center",
          "display": "flex",
          "gap": "0.5rem",
          "justifyContent": "center",
          "marginTop": "1rem",
        }
      }
    >
      <a
        href="https://twitter.com/mancoba_c"
        rel="noreferrer"
        target="_blank"
      >
        <svg
          aria-hidden={true}
          aria-label="Twitter"
          className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
          data-testid="TwitterIcon"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M22.46 6c-.77.35-1.6.58-2.46.69.88-.53 1.56-1.37 1.88-2.38-.83.5-1.75.85-2.72 1.05C18.37 4.5 17.26 4 16 4c-2.35 0-4.27 1.92-4.27 4.29 0 .34.04.67.11.98C8.28 9.09 5.11 7.38 3 4.79c-.37.63-.58 1.37-.58 2.15 0 1.49.75 2.81 1.91 3.56-.71 0-1.37-.2-1.95-.5v.03c0 2.08 1.48 3.82 3.44 4.21a4.22 4.22 0 0 1-1.93.07 4.28 4.28 0 0 0 4 2.98 8.521 8.521 0 0 1-5.33 1.84c-.34 0-.68-.02-1.02-.06C3.44 20.29 5.7 21 8.12 21 16 21 20.33 14.46 20.33 8.79c0-.19 0-.37-.01-.56.84-.6 1.56-1.36 2.14-2.23z"
          />
        </svg>
      </a>
      <a
        href="https://github.com/manq2010"
        rel="noreferrer"
        target="_blank"
      >
        <svg
          aria-hidden={true}
          aria-label="github"
          className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
          data-testid="GitHubIcon"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M12 1.27a11 11 0 00-3.48 21.46c.55.09.73-.28.73-.55v-1.84c-3.03.64-3.67-1.46-3.67-1.46-.55-1.29-1.28-1.65-1.28-1.65-.92-.65.1-.65.1-.65 1.1 0 1.73 1.1 1.73 1.1.92 1.65 2.57 1.2 3.21.92a2 2 0 01.64-1.47c-2.47-.27-5.04-1.19-5.04-5.5 0-1.1.46-2.1 1.2-2.84a3.76 3.76 0 010-2.93s.91-.28 3.11 1.1c1.8-.49 3.7-.49 5.5 0 2.1-1.38 3.02-1.1 3.02-1.1a3.76 3.76 0 010 2.93c.83.74 1.2 1.74 1.2 2.94 0 4.21-2.57 5.13-5.04 5.4.45.37.82.92.82 2.02v3.03c0 .27.1.64.73.55A11 11 0 0012 1.27"
          />
        </svg>
      </a>
      <a
        href="https://www.linkedin.com/in/mancoba-sihlongonyane/"
        rel="noreferrer"
        target="_blank"
      >
        <svg
          aria-hidden={true}
          aria-label="LinkedIn"
          className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
          data-testid="LinkedInIcon"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M19 3a2 2 0 0 1 2 2v14a2 2 0 0 1-2 2H5a2 2 0 0 1-2-2V5a2 2 0 0 1 2-2h14m-.5 15.5v-5.3a3.26 3.26 0 0 0-3.26-3.26c-.85 0-1.84.52-2.32 1.3v-1.11h-2.79v8.37h2.79v-4.93c0-.77.62-1.4 1.39-1.4a1.4 1.4 0 0 1 1.4 1.4v4.93h2.79M6.88 8.56a1.68 1.68 0 0 0 1.68-1.68c0-.93-.75-1.69-1.68-1.69a1.69 1.69 0 0 0-1.69 1.69c0 .93.76 1.68 1.69 1.68m1.39 9.94v-8.37H5.5v8.37h2.77z"
          />
        </svg>
      </a>
      <a
        href="mailto:manq2010@gmail.com"
        rel="noreferrer"
        target="_blank"
      >
        <span
          type="button"
        >
          <svg
            aria-hidden={true}
            className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
            data-testid="EmailIcon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M20 4H4c-1.1 0-1.99.9-1.99 2L2 18c0 1.1.9 2 2 2h16c1.1 0 2-.9 2-2V6c0-1.1-.9-2-2-2zm0 4-8 5-8-5V6l8 5 8-5v2z"
            />
          </svg>
        </span>
      </a>
    </div>
  </div>
  <div
    style={
      Object {
        "alignItems": "center",
        "borderTop": "1px solid #fff",
        "display": "flex",
        "gap": "1rem",
        "justifyContent": "space-around",
      }
    }
  >
    <div
      style={
        Object {
          "margin": "1rem 0 1rem 0",
        }
      }
    >
      <p>
        Created by
         
        <em>
          Mancoba Sihlongonyane
        </em>
      </p>
    </div>
    <div>
      <p>
        Â© Copyright
         
        2023 -
         
        2023
      </p>
    </div>
  </div>
</div>
`;

exports[`Not-found page is rendered matches snapshot 1`] = `
<div
  className="sc-bcXHqe efAyRB"
>
  <h3>
    404!
  </h3>
  <p>
    Page
     
    <em>
      Not
    </em>
     
    Found!
  </p>
</div>
`;
